LaTeX/LaTeX.sublime-syntax

```diff
@@ -1656,3 +1657,137 @@ contexts:
        3: punctuation.definition.group.brace.begin.latex
        4: entity.name.function.frame.latex
        5: punctuation.definition.group.brace.end.latex
+
+  # support for array package
+  array:
+    - match: |-
+        (?x)
+        (
+          (\\)newcolumntype
+        )
+        (?:
+          (\{)
+          (
+            (?:\\[A-Za-z@]+)
+            | (?:[A-Za-z@])
+          )
+          (\})
+        )
+        (?:(\[)(?:[^\]]*)(\]))*
+        (\{)
+      captures:
+        1: support.function.newcolumntype.latex
+        2: punctuation.definition.backslash.latex
+        3: punctuation.definition.group.brace.begin.latex
+        4: support.function.latex entity.name.newcolumntype.latex
+        5: punctuation.definition.group.brace.end.latex
+        6: punctuation.definition.group.bracket.begin.latex
+        7: punctuation.definition.group.bracket.end.latex
+        8: punctuation.definition.group.brace.begin.latex
+      push:
+        - meta_include_prototype: false
+        - meta_scope: meta.function.newcolumntype.latex
+        - match: '\}'
+          scope: punctuation.definition.group.brace.end.latex
+          pop: true
+        - include: array-preamble
+
+
+    - match: |-
+        (?x)
+        ((\\)begin)(\{)(tabular)(\})
+        (?:(\[)(?:b|c|t)(\]))?
+        (\{)
+      captures:
+        1: support.function.begin.latex keyword.control.flow.begin.latex
+        2: punctuation.definition.backslash.latex
+        3: punctuation.definition.group.brace.begin.latex
+        4: variable.parameter.function.latex
+        5: punctuation.definition.group.brace.end.latex
+        6: punctuation.definition.group.bracket.begin.latex
+        7: punctuation.definition.group.bracket.end.latex
+        8: punctuation.definition.group.brace.begin.latex
+      push:
+        - meta_scope: meta.environment.tabular.latex
+        - match: '(?<=\{)'
+          scope: punctuation.definition.group.brace.begin.latex
+          push:
+            - meta_include_prototype: false
+            - meta_scope: meta.function.column-spec.latex
+            - match: '\}'
+              pop: true
+            - include: array-preamble
+        - match: '((\\)end)(\{)(tabular)(\})'
+          captures:
+            1: support.function.end.latex keyword.control.flow.end.latex
+            2: punctuation.definition.backslash.latex
+            3: punctuation.definition.group.brace.begin.latex
+            4: variable.parameter.function.tabular.latex
+            5: punctuation.definition.group.brace.end.latex
+          pop: true
+        - include: main
+
+  array-preamble:
+    - match: 'l|r|c'
+      scope: keyword.other.column-type.latex
+
+    - match: 'p|m|b'
```

Comment:
We may change this to `'(?:p|m|b)(?=\s*\{)'` to ensure it is followed by a brace, otherwise the syntax breaks for a missing `{...}`.